internal/controllers/analyzer/analyzer.go

```diff
@@ -457,3 +468,57 @@ func (a *Analyzer) isWarning(err string) bool {
 		strings.Contains(err, messages.MsgWarnBrakemanNotRubyOnRailsProject) ||
 		strings.Contains(err, messages.MsgWarnGemfileIsRequiredForBundler)
 }
+
+// getAllVulnerabilitiesWithDetailsJoined will add the default separator of the details and will check if the hash
+// already exists in list of the vulnerabilities, if is duplicated, so, it will only join both details.
+// nolint:funlen // Breaking this function will make it more confusing
+func (a *Analyzer) getAllVulnerabilitiesWithDetailsJoined() (newAV []analysis.AnalysisVulnerabilities) {
+	for indexAV := range a.analysis.AnalysisVulnerabilities {
+		exists := false
+		for index := range newAV {
+			if a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.VulnHash ==
+				newAV[index].Vulnerability.VulnHash {
+				exists = true
+				newAV[index].Vulnerability.Details = fmt.Sprintf("%s\n         %s%s",
+					newAV[index].Vulnerability.Details,
+					detailsHeaderText,
+					a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details)
+				break
+			}
+		}
+		if !exists {
+			a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details = fmt.Sprintf("%s%s",
+				detailsHeaderText, a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details)
+			newAV = append(newAV, a.analysis.AnalysisVulnerabilities[indexAV])
+		}
+	}
+	return
+}
+
+// setCounterOfDetailsDuplicated will verify how details exists looking the separator detailsHeaderText param and
+// will update to add a counter of the details in this vulnerability.
+// nolint:funlen,gocyclo // Breaking this function will make it more confusing
+func (a *Analyzer) setCounterOfDetailsDuplicated(
+	manyToMany []analysis.AnalysisVulnerabilities,
+) []analysis.AnalysisVulnerabilities {
+	for indexMM := range manyToMany {
+		newDetails := ""
+		detailEmptySplit := strings.Split(manyToMany[indexMM].Vulnerability.Details, detailsHeaderText)
+		detailToSetSplit := make([]string, 0)
+		for _, d := range detailEmptySplit {
+			if d != "" {
+				detailToSetSplit = append(detailToSetSplit, d)
+			}
+		}
+		for indexDS, detail := range detailToSetSplit {
+			lastChar := "\n"
+			if indexDS+1 == len(detailToSetSplit) {
+				lastChar = ""
+			}
+			newDetails = fmt.Sprintf("%s(%v/%v) %s%s%s", newDetails, indexDS+1, len(detailToSetSplit),
+				detailsHeaderText, detail, lastChar)
```

Comment:
I did not understand

Reply:
Instead of concatenate the newDetails using the fmt.Sprintf we can use a string builder:
```go
details := bytes.NewStringBuffer("")
fmt.Fprintf(details, "(%v/%v) %s%s%s", indexDS+1, len(detailToSetSplit), detailsHeaderText, detail, lastChar)
```

Reply:
it doesn't make sense here, because we are still modifying the analysis object, only in print_results we will have the print for the end user, and doing this work in the print_results package I don't know if it would be a good idea

Reply:
I'm referring about the concatenation that occur on newDetails variable, not analysis object. 
```go
newDetails := bytes.newBufferString("")
...

for indexDS, detail := range detailToSetSplit {
	...
	fmt.Fprintf(newDetails, "(%v/%v) %s%s%s", newDetails, indexDS+1, len(detailToSetSplit), detailsHeaderText, detail, lastChar)
	...
}
currentAV.Vulnerability.Details = newDetails.String()
```

Reply:
i don't like use this... but will change :+1: